src/rules/no-unsafe-values.js

```diff
@@ -25,6 +31,48 @@ export default {
 						loc: node.loc,
 						messageId: "unsafeNumber",
 					});
+				} else {
+					const txt = context.sourceCode.getText(node);
+					if (node.value === 0) {
+						// If the value has been rounded down to 0, but there was some fraction
+						// or non-zero part before an e-, this is a very small number that doesn't
+						// fit inside an f64.
+						for (const r of SMALL_NUMBERS) {
+							const m = txt.match(r);
+							// If the digits are all 0, it's ok.
+							if (m?.[1]?.match(/[1-9]/u)) {
+								context.report({
+									loc: node.loc,
+									messageId: "unsafeNumber",
+								});
+								break;
+							}
+						}
+					} else if (!txt.match(/[.e]/iu)) {
+						// Intended to be an integer
+						if (
+							node.value > Number.MAX_SAFE_INTEGER ||
+							node.value < Number.MIN_SAFE_INTEGER
+						) {
+							context.report({
+								loc: node.loc,
+								messageId: "unsafeInteger",
+							});
+						}
+					} else {
+						// Floating point.  Check for subnormal.
+						const ab = new ArrayBuffer(8);
+						const dv = new DataView(ab);
+						dv.setFloat64(0, node.value, false);
+						const bi = dv.getBigUint64(0, false);
+						// Subnormals have an 11-bit exponent of 0 and a non-zero mantissa.
+						if ((bi & 0x7ff0000000000000n) === 0n) {
```

Comment:
Expanding variable names.
```suggestion
						const buffer = new ArrayBuffer(8);
						const view = new DataView(buffer);
						view.setFloat64(0, node.value, false);
						const bigInt = view.getBigUint64(0, false);
						// Subnormals have an 11-bit exponent of 0 and a non-zero mantissa.
						if ((bigInt & 0x7ff0000000000000n) === 0n) {
```